frontends/web/src/routes/account/info/buyReceiveCTA.tsx

```diff
@@ -48,22 +55,31 @@ export const BuyReceiveCTA = ({ code, unit, balanceList }: TBuyReceiveCTAProps)
       <h3 className="subTitle">{t('accountInfo.buyCTA.information.start')}</h3>
       <div className={styles.container}>
         {balanceList && <Button primary onClick={onReceiveCTA}>{formattedUnit ? t('receive.title', { accountName: formattedUnit }) : t('receive.title', { accountName: t('buy.info.crypto') })}</Button>}
-        <Button primary onClick={onBuyCTA}>{formattedUnit ? t('accountInfo.buyCTA.buy', { unit: formattedUnit }) : t('accountInfo.buyCTA.buyCrypto')}</Button>
+        {exchangeBuySupported && <Button primary onClick={onBuyCTA}>{formattedUnit ? t('accountInfo.buyCTA.buy', { unit: formattedUnit }) : t('accountInfo.buyCTA.buyCrypto')}</Button>}
       </div>
     </div>);
 };
 
 
-export const AddBuyReceiveOnEmptyBalances = ({ balances }: {balances?: Balances}) => {
-  if (balances === undefined) {
+export const AddBuyReceiveOnEmptyBalances = ({ balances, accounts }: TAddBuyReceiveOnEmpyBalancesProps) => {
+
+  const [supportedAccounts, setSupportedAccounts] = useState<IAccount[]>();
+
+  useEffect(() => {
+    getExchangeSupportedAccounts(accounts).then(setSupportedAccounts);
+  }, [accounts]);
+
+  if (balances === undefined || supportedAccounts === undefined) {
```

Comment:
Looks like  supportedAccounts could also be an empty array 🤔  

shouldn't it also check `|| supportedAccounts.length > 0` or similar?

Reply:
Although I don't think checking .length > 0 matters though as this component won't get rendered anyway  🤔 I put an undefined check there because of the async fetch, so when it's loading it's short-circuited (thus return `null`) instead of it keeps going.

Reply:
Actually, I don't think we should check for `.length > 0` here because if it's === 0 then it'll still have to return the BuyReceiveCTA but `exchangeBuySupported` is `false` which we're already doing. The `undefined` here signifies that `getExchangeSupportedAccounts ` is still loading thus I short-circuit here. 🙂

Reply:
ah yeah you are totally right. thanks 👍